packages/context/src/binding.ts

```diff
@@ -128,6 +133,11 @@ export type BindingTag = TagMap | string;
  */
 export type BindingTemplate<T = unknown> = (binding: Binding<T>) => void;
 
+type ValueGetter<T> = (
+  ctx: Context,
+  options: ResolutionOptions,
+) => ValueOrPromise<T | undefined>;
```

Comment:
I am really not comfortable with this change and need more time to research possible ramifications.

If you can change this line to return `ValueOrPromise<T>` then this PR could be landed sooner.

```suggestion
) => ValueOrPromise<T>;
```